### PR TITLE
fix: enable editing charts with errors

### PR DESCRIPTION
**Pull request by magnew**
Closes: #8016 

### Description:

We were not showing the chart editing controls when a chart errored out. This adds the edit button to a tile with errors. 

Writing this made me want to clean up the tile hierarchy more. There are a lot of wrappers that probably don't all need to be in there. 

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/DashboardTiles/EditChartMenuItem.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    const dashboard = useDashboardContext((c) => c.dashboard);
+
+    const { storeDashboard } = useDashboardStorage();
+
+    const { projectUuid, dashboardUuid } = useParams<{
+        projectUuid: string;
+        dashboardUuid: string;
+    }>();
+
+    const userCanMangeExplore = user.data?.ability?.can('manage', 'Explore');
```
Comment (joaoviana) [inline packages/frontend/src/components/DashboardTiles/EditChartMenuItem.tsx:27]:
```suggestion
    const userCanManageExplore = user.data?.ability?.can('manage', 'Explore');
```

**Review by joaoviana — COMMENTED** [review]
one typo

Testing in Render env now 👍

**Review by joaoviana — CHANGES_REQUESTED** [review]
When on this dashboard: https://lightdash-pr-8022.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/dashboards/f5ce5f73-0bea-4f8b-9b4a-99404bf52af7/view

I deleted chart tile 2 and didn't get any option to delete it on the menu. 
image

**Comment by magnew** [comment]
> When on this dashboard: https://lightdash-pr-8022.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/dashboards/f5ce5f73-0bea-4f8b-9b4a-99404bf52af7/view
> 
> I deleted chart tile 2 and didn't get any option to delete it on the menu. image


Thanks! good catch!

**Review by joaoviana — APPROVED** [review]
Tested:
* saved charts and can edit and delete
* charts that were broken: can edit(change the chart) or remove them **in edit mode**

**Pull request merged**